projects/valtimo/dossier/src/lib/components/dossier-detail/tab/progress/progress.component.ts

```diff
@@ -14,43 +14,75 @@
  * limitations under the License.
  */
 
-import {Component, OnInit} from '@angular/core';
+import {Component} from '@angular/core';
 import {ActivatedRoute} from '@angular/router';
-import {DocumentService} from '@valtimo/document';
-import {ProcessDocumentInstance} from '@valtimo/document';
+import {DocumentService, ProcessDocumentInstance} from '@valtimo/document';
+import {BehaviorSubject, combineLatest, map, Observable, of, switchMap, tap} from 'rxjs';
+import {ListItem} from 'carbon-components-angular/dropdown';
 
 @Component({
   selector: 'valtimo-dossier-detail-tab-progress',
   templateUrl: './progress.component.html',
   styleUrls: ['./progress.component.css'],
 })
-export class DossierDetailTabProgressComponent implements OnInit {
-  public processDocumentInstances: ProcessDocumentInstance[];
-  public selectedProcessInstanceId: string;
-  public readonly documentId: string;
+export class DossierDetailTabProgressComponent {
+  private readonly processDocumentInstances$: Observable<Array<ProcessDocumentInstance>> = of(
+    this.route.snapshot.paramMap.get('documentId')
+  ).pipe(
+    switchMap(documentId => this.documentService.findProcessDocumentInstances(documentId)),
```

Comment:
this.route.paramMap is an observable by default, no need to make it on your own from the snapshot. Check dossier-detail.component.ts on the canAssign$ attribute

Reply:
Changed